### PR TITLE
[backport] Speedup Upsampling2D on CPU

### PR DESCRIPTION
**Pull request by mitmul**
This PR backports #3316

### CODE REVIEW

**Comment by niboshi** [comment]
It includes changes not existed in the original PR.
Are they needed?
If so, let's backport those changes first (#2978).
Otherwise, it may be inconsistent with other part of the code.

**Comment by mitmul** [comment]
Thanks, I didn't notice that. will fix them

File: chainer/functions/pooling/upsampling_2d.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
+import numpy
```
Comment (niboshi) [inline chainer/functions/pooling/upsampling_2d.py]: Please revert this, too.

Comment (mitmul) [reply chainer/functions/pooling/upsampling_2d.py]: Fixed

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**